hwmv2: Added Board Support for Starfive VisionFive2 RISC-V Single Board Computer

pfarwsi opened the pull request:
Modified the patches according to HWMV2 as per #69359. These list of patches contain Board support files for running Zephyr on the Starfive VisionFive2 RISC-V Single Board Computer based on JH7110 SOC. It's a basic support, no drivers other than UART have been tested.

tejlmand committed:
doc: update board porting guide to match new hardware model

Update board porting guide to the design changes in the new hw model.

New section added:
- board.yml file description

Updated sections:
- Board on SoCs example table updated to new format and extra examples
  added.
- Board directory description and its files
- Writing devicetree file section updated to match new lookup patterns
- Writing Kconfig file section updated to match new lookup patterns
- Board revision handling updated

In getting started guide a note has been added with short description
of the new board and board identifier scheme.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

nordicjm committed:
soc: snps_qemu: Port to HWMv2

Ports the snps_qemu SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arc: qemu_arc: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: snps_nsim: Port to HWMv2

Ports the snps_nsim SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arc: nsim: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: snps_arc_hsdk: Port to HWMv2

Ports the snps_arc_hsdk SoC configuration to hardware model
version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arc: hsdk: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: snps_emsk: Port to HWMv2

Ports the snps_emsk SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arc: em_starterkit: Convert to v2

Converts the board to hwmv2, documentation for this board needs to
be vastly updated

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
tests: Update board names for hwmv2

Updates tests that use board names which have changed with boards v2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: snps_emsdp: Port to HWMv2

Ports the snps_emsdp SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arc: emsdp: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: snps_arc_iot: Port to HWMv2

Ports the snps_arc_iot SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arc: iotdk: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: snps_arc_hsdk4xd: Port to HWMv2

Ports the snps_arc_hsdk4xd SoC configuration to hardware model
version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arc: hsdk4xd: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: psoc6 and psoc_6: Port to HWMv2

Ports the psoc6 and psoc_6 SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: cy8ckit_062_ble: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: cy8ckit_062_wifi_bt: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: cy8ckit_062s4: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: cy8cproto_062_4343w: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: cy8cproto_063_ble: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: xmc_4xxx: Port to HWMv2

Ports the xmc_4xxx SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: xmc45_relax_kit: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: xmc47_relax_kit: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
manifest: Update hal_infineon

Updates hal_infineon with changes needed for hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

fkokosinski committed:
soc/sifive/sifive_freedom: add SiFive Freedom E310 SoC

This commit adds support for the SiFive Freedom E310 SoC for the Zephyr
Hardware Model v2.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

fkokosinski committed:
boards: convert SiFive HiFive1 to Zephyr HWMv2

This commit converts the SiFive HiFive1 board (`hifive1` target) to the
Zephyr Hardware Model v2.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

fkokosinski committed:
soc/sifive/sifive_freedom: add SiFive Freedom U540 SoC

This commit adds support for the SiFive Freedom U540 SoC for the Zephyr
Hardware Model v2.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

fkokosinski committed:
boards: convert SiFive HiFive Unleashed to Zephyr HWMv2

This commit converts the SiFive HiFive Unleashed board (`hifive_unleashed`
target) to the Zephyr Hardware Model v2.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

fkokosinski committed:
soc/sifive/sifive_freedom: add SiFive Freedom FU740 SoC

This commit adds support for the SiFive Freedom U740 SoC for the Zephyr
Hardware Model v2.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

fkokosinski committed:
boards: convert SiFive HiFive Unmatched to Zephyr HWMv2

This commit converts the SiFive HiFive Unmatched board (`hifive_unmatched`
target) to the Zephyr Hardware Model v2.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

fkokosinski committed:
boards: convert SparkFun RED-V Things Plus to Zephyr HWMv2

This commit converts the SparkFun RED-V Things Plus board
(`sparkfun_red_v_things_plus` target) to the Zephyr Hardware Model v2.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

57300 committed:
cmake: modules: boards: Fix BOARD_ALIAS

`--format` is not a valid argument to `list_boards.py`.

Signed-off-by: Grzegorz Swiderski <grzegorz.swiderski@nordicsemi.no>

57300 committed:
cmake: modules: boards: Enhance board aliases for HWMv2

To address concerns about lengthy board identifiers in HWMv2, the
proposal is to apply the existing BOARD_ALIAS feature, like so:

   set(<alias>_BOARD_ALIAS <board>/<soc>)

It should then be possible to build with either:

   -DBOARD=<alias>            # expands to <board>/<soc>
   -DBOARD=<alias>/<variant>  # expands to <board>/<soc>/<variant>

However, this wouldn't work out of the box. A board alias can only be
expanded to a board name, without revision or identifier, because the
alias substitution happens after having parsed BOARD as user input -
namely, into BOARD (name), BOARD_REVISION, and BOARD_IDENTIFIER.

Furthermore, this means that in the legacy model, it was possible to
build with `-DBOARD=<alias>@<revision>`, and it would resolve to the
actual board name + revision.

To support both the old and new use cases, we can parse the alias just
like BOARD itself, then concatenate their identifiers as shown above.
Adding a revision works as before, but now it is also possible for the
alias to set its own revision. In this example:

   set(<alias>_BOARD_ALIAS <board>@<rev-A>/<soc>/<variant>)

`<rev-A>` is treated as the default revision, and it can be overridden:

   -DBOARD=<alias>          # expands to <board>@<rev-A>/<soc>/<variant>
   -DBOARD=<alias>@<rev-B>  # expands to <board>@<rev-B>/<soc>/<variant>

Signed-off-by: Grzegorz Swiderski <grzegorz.swiderski@nordicsemi.no>

manuargue-nxp committed:
boards: nxp: ucans32k1sic: convert to hwmv2

Convert ucans32k1sic board to hardware model v2.

Signed-off-by: Manuel Argüelles <manuel.arguelles@nxp.com>

manuargue-nxp committed:
boards: nxp: mr_canhubk3: convert to hwmv2

Convert mr_canhubk3 board to hardware model v2.

Signed-off-by: Manuel Argüelles <manuel.arguelles@nxp.com>

nashif committed:
boards: qemu: move qemu_x86 -> x86

Just follow others and drop qemu_ from the name, we have that in the
folder already.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
x86: remove legacy ia32, use atom instead

ia32 is legacy and is just an atom, so deduplicate and use generic
atom soc instead.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
x86: atom: remove soc.h with unused content

None of the code in soc.h is being used anywhere, so remove it and
remove include from various places.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

57300 committed:
cmake: modules: boards: Fix board deprecation for HWMv2

Deprecating boards comes with similar challenges as with board aliases,
since BOARD_DEPRECATED is also set after parsing BOARD as user input.

With this patch, a deprecated board can be properly translated to a
board with identifiers. This opens up the possibility of recording all
legacy board names in `boards/deprecated.cmake`, such as:

   set(mps2_an521_remote_DEPRECATED mps2/an521/cpu1)

Unlike with aliases, though, there are additional restrictions for
building with BOARD=<deprecated>, which only makes this feature suitable
for deprecating v1 boards:

 * BOARD=<deprecated>/<identifier> is never allowed.
 * BOARD=<deprecated>@<revision> is not allowed if the old board already
   corresponds to a new board revision:

   set(<deprecated>_DEPRECATED <new-board>@<new-revision>)

Future enhancements will be needed for deprecating v2 boards.

Signed-off-by: Grzegorz Swiderski <grzegorz.swiderski@nordicsemi.no>

Commit:
hwmv2: Global fix of lpc54114_m4 overlay and conf files

Global fixup of overlay and conf files

Signed-off-by: David Leach <david.leach@nxp.com>

manuargue-nxp committed:
soc: nxp: convert NXP S32 family to hwmv2

Convert NXP S32 family to hardware model v2.

Signed-off-by: Manuel Argüelles <manuel.arguelles@nxp.com>

fkokosinski committed:
MAINTAINERS: update RISC-V arch area paths

This commit updates the paths assigned to the RISC-V area of maintenance to
include targets based on the SiFive Freedom SoC family.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

fabiobaltieri committed:
soc: convert microchip_mec to hwmv2

Convert the Microchip MEC soc/ to hardware model v2.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri committed:
boards: convert all microchip MEC boards to hwmv2

Convert mec1501modular_assy6885, mec15xxevb_assy6853,
mec172xevb_assy6906 and mec172xmodular_assy6930 to hwmv2.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri committed:
boards: adjust few boards/ paths

Adjust few paths in the board document and maintainer file to reflect
the new file location.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

golowanow committed:
boards: arm64: intel_socfpga: Move boards to subdirectories

Adjust boards directory structure to SoC structure:

 `boards/intel/intel_socfpga_agilex_socdk` ->
     `boards/intel/intel_socfpga/agilex_socdk`
 `boards/intel/intel_socfpga_agilex5_socdk` ->
     `boards/intel/intel_socfpga/agilex5_socdk`

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
MAINTAINERS: intel_socfpga: Adjust to HWMv2 move

Adjust intel/intel_socfpga to HWMv2 move.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

nashif committed:
boards: x86: remove qemu_x86_tiny_768

Having a board to just measure coverage for certain features is a bit of
any overkill. We will instead add overlays to the specific tests this
board mostly covers.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
soc: x86/atom: imply mmu, do not select it

Board should be able to deselect mmu, so imply instead of strict
selection.

userspace is selected for atom already on the CPU level, so do not
reselect.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nordicjm committed:
boards: synopsys: Add documentation index

Adds a documentation index file

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
maintainers: Update synopsys entries

Updates entries to account for hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: cypress: Add documentation index

Adds documentation index file

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
maintainers: Update paths for hwmv2

Updates paths which have changed in hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: infineon: Add documentation index

Adds documentation index file

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

fkokosinski committed:
boards: convert SiFive HiFive1 Rev. B to Zephyr HWMv2

This commit converts the SiFive HiFive1 Rev. B board (`hifive1_revb`
target) to the Zephyr Hardware Model v2.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

nashif committed:
ci: doc: always report status

Generate status that can be used for enforcing branch checks even if
some jobs get skipped.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
ci: twister status check

Generate one single status for all jobs and also report success when
twister run is being skipped to be able to make this a required check.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

golowanow committed:
soc: arm64: intel_socfpga: hwmv2: Rename SoCs

Rename SoC for better HWMv2 use:
  `intel_socfpga_agilex`  --> `agilex`
  `intel_socfpga_agilex5` --> `agilex5`

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

nordicjm committed:
soc: soc_legacy: mips: Remove out file

Removes a file that is no longer used

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: gaisler: Move Kconfig file

Moves a Kconfig file from hwmv1 to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: silabs_exx32: Port to HWMv2

Ports the SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: efm32gg_sltb009a: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: efm32gg_slwstk6121a: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: efm32gg_stk3701a: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: efm32hg_slstk3400a: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: efm32pg_stk3401a: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: efm32wg_stk3800: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: efr32xg24_dk2601b: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: efr32mg_sltb004a: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: efr32_thunderboard: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
manifest: Update hal_silabs

Updates the repo to include hwmv2 changes

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: efm32pg_stk3402a: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: efr32_radio: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: silab: Add documentation index file

Adds a file with the vendor name for documentation indexing

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
tests: Update board names for hwmv2

Updates names used in tests for hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

fkokosinski committed:
soc/riscv: convert the QEMU virt RISCV-32 SoC to HWMv2

This commit converts the virt SoC (generic virt machine) to the Zephyr
HWMv2.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

erwango committed:
boards: stm32wb: Restore missing .defconfig files

Kconfig.defconfig files has been lost during migration.
Put them back.

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

Commit:
soc: nxp: convert LPC SOC family to hardware model V2

Move LPC family to HWMv2

Signed-off-by: David Leach <david.leach@nxp.com>

soc: nxp: convert LPC SOC family to hardware model V2

Move LPC family to HWMv2

Signed-off-by: David Leach <david.leach@nxp.com>

Commit:
boards: nxp:  convert lpcxpresso11u68 to hwmv2

Convert lpcxpresso11u68

Signed-off-by: David Leach <david.leach@nxp.com>

EmilioCBen committed:
hwmv2: lpc: updated overlay and conf files in samples and tests

Fixup for LPC54114 and LPC55S69

Signed-off-by: Emilio Benavente <emilio.benavente@nxp.com>
Signed-off-by: David Leach <david.leach@nxp.com>

Commit:
boards: nxp: convert lpcxpresso51u68 to hwmv2

Convert lpcxpresso51u68

Signed-off-by: David Leach <david.leach@nxp.com>

Commit:
boards: nxp: convert lpcxpresso55s06 to hwmv2

Convert lpcxpresso55s06

Signed-off-by: David Leach <david.leach@nxp.com>

Commit:
boards: nxp: convert lpcxpresso55s16 to hwmv2

Convert lpcxpresso55s16

Signed-off-by: David Leach <david.leach@nxp.com>

Commit:
boards: nxp: convert lpcxpresso55s28 to hwmv2

Convert lpcxpresso55s28

Signed-off-by: David Leach <david.leach@nxp.com>

Commit:
boards: nxp: convert lpcxpresso55s36 to hwmv2

Convert lpcxpresso55s36

Signed-off-by: David Leach <david.leach@nxp.com>

Commit:
boards: nxp: convert lpcxpresso54114 to hwmv2

Convert lpcxpresso54114

Signed-off-by: David Leach <david.leach@nxp.com>

Commit:
boards: nxp: convert lpcxpresso55s69 to hwmv2

Convert lpcxpresso55s69

Signed-off-by: David Leach <david.leach@nxp.com>

Commit:
hwmv2: soc: Port IMXRT family to HWMV2

Port IMXRT family to HWMV2, including series:
- RT11XX
- RT10XX
- RT6XX

Not including RT5XX

Co-authored-by: Declan Snyder <declan.snyder@nxp.com>
Co-authored-by: Daniel DeGrasse <daniel.degrasse@nxp.com>
Co-authored-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>
Co-authored-by: David Leach <david.leach@nxp.com>
Co-authored-by: Yves Vandervennet <yves.vandervennet@nxp.com>
Co-authored-by: Emilio Benavente <emilio.benavente@nxp.com>

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
hwmv2: boards: Convert IMXRT boards

Convert IMXRT boards except RT595

Co-authored-by: Declan Snyder <declan.snyder@nxp.com>
Co-authored-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
hwmv2: nxp: update MAINTAINER paths for hwmv2

Update for hwmv2

Signed-off-by: David Leach <david.leach@nxp.com>

danieldegrasse committed:
soc: nxp: rt5xx: convert RT5xx SOC to HWMv2

Convert NXP iMX RT5xx SOC to hardware model V2. This core is dual
architecture, so both the ARM Cortex M33 and Xtensa Fusion F1 DSP have
been unified within one SOC port folder.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
boards: nxp: convert mimxrt595_evk to hardware model v2

Convert mimxrt595_evk to hardware model v2. Both the CM33 and Fusion F1
DSP devicetrees are moved to reside in the same board folder

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

nordicjm committed:
boards: Fix invalid documentation links

Fixes issues with links in documentation

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: Use zephyr_file for file links

Replaces normal text with :zephyr_file: blocks

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

JiafeiPan committed:
hwmv2: move imx93 soc to V2

Port NXP imx93 SoC to hwm V2.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
hwmv2: move imx93 EVK board to V2

Port EVK board for NXP imx93 to hwm V2.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

fkokosinski committed:
boards: convert QEMU RISC-V 32 bit board to Zephyr HWMv2

This commit converts the QEMU RISCV-V 32 bit board to Zephyr HWMvW. This
includes the following former targets:
* qemu_riscv32
* qemu_riscv32_smp
* qemu_riscv32_xip

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

fkokosinski committed:
boards: convert QEMU RISC-V 64 bit board to Zephyr HWMv2

This commit converts the QEMU RISCV-V 64 bit board to Zephyr HWMvW. This
includes the following former targets:
* qemu_riscv64
* qemu_riscv64_smp

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

fkokosinski committed:
boards: convert QEMU RV32E board to Zephyr HWMv2

This commit converts the QEMU RV32E board to Zephyr HWMvW. This includes
the following former target: qemu_riscv32e.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

nordicjm committed:
modules: tf-m: Rename from nordic_nrf to nordic

Aligns with the vendor prefix

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
various: Change SOC_FAMILY_NRF to SOC_FAMILY_NORDIC_NRF

Fixes the name to include the vendor in it

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: andes: Rename to andestech

Aligns with vendor prefix

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: beagleboard: Change to beagle

Aligns with vendor prefix

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: broadcom: Rename to brcm

Aligns with vendor prefix

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: cadence: Rename to cdns

Aligns with vendor prefix

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: laird_connect: Rename to lairdconnect

Aligns with vendor prefix

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: seeed_studio: Rename to seeed

Aligns with vendor prefix

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: space_cubics: Rename to sc

Aligns with vendor prefix

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: ublox: Rename to u-blox

Aligns with vendor prefix

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: altera: Rename to altr

Aligns with vendor prefix

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: andes: Rename to andestech

Aligns with vendor prefix

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: broadcom: Rename to brcm

Aligns with vendor prefix

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: cadence: Rename to cdns

Aligns with vendor prefix

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: xenvm: Rename to xen

Aligns with vendor prefix

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: xilinx: Rename to xlnx

Aligns with vendor prefix

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
MAINTAINERS: Update for renamed folders

Updates for folders renamed to have their vendor prefix names

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
tests: Fix board names

Fixes some invalid board names

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

Commit:
hwmv2: Convert Seagate Faze board to hwmv2

Convert Faze board

Signed-off-by: David Leach <david.leach@nxp.com>

tejlmand committed:
scripts: ci: introduce soc name check in check_compliance

soc.yml files define SoC names which are used in board.yml.
All SoC names and directories are exported to the build system and
can be referenced using the SoC name as identifier.

Kconfig defines a CONFIG_SOC setting with the same name which can be
used in build system and is selected by the board.

Thus the CONFIG_SOC value can be used to lookup the details of the SoC.

This commit introduces a new compliance check which ensures the SoC name
and the CONFIG_SOC name value are in sync.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

tejlmand committed:
soc: align CONFIG_SOC values to match soc.yml names

This commit aligns CONFIG_SOC values for Nordic nRF SoCs to the names
defined in soc.yml.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

JiafeiPan committed:
hwmv2: move imx8m soc for a-core to V2

Port NXP imx8m serial SoC to hwm V2, it includes imx8mm, imx8mn and
imx8mp.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
hwmv2: move imx8mm EVK board to V2

Port EVK board for NXP imx8mm to hwm V2.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
hwmv2: move imx8mp EVK board to V2

Port EVK board for NXP imx8mp to hwm V2.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
hwmv2: move imx8mn EVK board to V2

Port EVK board for NXP imx8mn to hwm V2.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

manuargue-nxp committed:
boards: nxp: s32z2xxdc2: convert to hwmv2

Convert `s32z270dc2` boards to hardware model v2. The board has been
renamed to `s32z2xxdc2` to be able to support in the future other
SoCs from this series that can also work on this board.

Signed-off-by: Manuel Argüelles <manuel.arguelles@nxp.com>

nordicjm committed:
boards: nordic: nrf5340dk: Fix board names

Updates to new board names

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
tests: kernel: tickless: Remove old board name

Removes a duplicated old board name

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

erwango committed:
soc: stm32: Cleanup Kconfig.defconfig files

SOC related symbols are defined in Kconfig.soc

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

erwango committed:
soc: stm32: Rename Kconfig SOC_SERIES symbols

SOC symbol is expected to match SOC_SERIES_FOO symbol.

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

tejlmand committed:
soc: ti: define SOC name in Kconfig

Add Kconfig SoC name am6234 which matches the SoC name in soc.yml.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

tejlmand committed:
soc: mec: align SoC names defined in soc.yml to Kconfig SOC setting

This commit aligns the SoC names for mec SoCs defined in soc.yml to
Kconfig SOC setting.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

tejlmand committed:
soc: arm: align SoC names defined in soc.yml to Kconfig SOC setting

This commit aligns the SoC names for arm an547 SoCs defined in soc.yml
to Kconfig SOC setting.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

tejlmand committed:
soc: broadcom: align SoC names defined in soc.yml to Kconfig SOC setting

This commit aligns the SoC names for bcm58402 SoCs defined in soc.yml
to Kconfig SOC setting.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

erwango committed:
soc: stm32: Rename series path

Since symbol is used by CMake to locate matching series folder, rename
folders.

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

nordicjm committed:
soc: cypress: Add protection guard to file

Adds a protection guard to prevent Kconfigs leaking

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

tejlmand committed:
boards: nrf51dk/dongle: change SoC to nRF51822

The physical nRF51dk / nRF51dongle hardware contains a nRF51422 SoC.
In Zephyr, only the nRF51822 SoC, is implemented.

In Zephyr, the nRF51422 SoC is build as a nRF51822, therefore use the
nRF51822 SoC model for those kits instead on the non-existing nRF51422.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

nordicjm committed:
soc: xilinx: zync7000: Remove xilinx from soc series name

Removes the vendor name from a Kconfig symbol

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

golowanow committed:
soc: intel_adsp_cavs: hwmv2: Align SOC_SERIES name and value

Align 'intel_adsp_cavs` SoC Series name and value to match
new HWMv2 compliance check.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
soc: intel_adsp: hwmv2: Align SOC_SERIES_INTEL_ACE name and value

Align `ace` to 'intel_adsp_ace` SoC Series name and value to match
the new HWMv2 compliance check, also renaming:

  SOC_SERIES_INTEL_ACE --> SOC_SERIES_INTEL_ADSP_ACE

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

nordicjm committed:
scripts: west_commands: completion: Add hwmv2 complete to bash

Adds hwmv2 board completetion support to bash

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
scripts: west: list_boards: Fix hwmv2 output

Fixes the output of list_boards to be in csv format rather than
python array format

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nashif committed:
twister: normalize platform name when storing files/data

Convert slashes into underscores to allow saving of data related to
platforms on disk.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nordicjm committed:
scripts: west_commands: completion: Add hwmv2 complete to zsh

Adds hwmv2 board completetion support to zsh

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
scripts: west_commands: completion: Add hwmv2 complete to fish

Adds hwmv2 board completetion support to fish

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nashif committed:
twister: tests: test with slashes in platform names

Use mps2/an385 for testing the hwmv2 board names.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nordicjm committed:
boards: nordic: nrf9160dk: Fix missing nrf52840 config

Fixes an issue with missing configuration from the nrf52840 on
the nrf9160dk

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

57300 committed:
snippets: Initial HWMv2 support

Very simple adaptation to directly replace the use of HWMv1 board names
with the new identifiers:

   boards:
     <name>/<identifier>:
       append: ...

All snippets in-tree are aligned with this format where needed.

Since the `boards` key supports regex, appends can also be applied to
multiple targets grouped by board name, SoC, CPU cluster, or variant.
Later, this can be followed up with native support for these items in
the snippet schema.

Signed-off-by: Grzegorz Swiderski <grzegorz.swiderski@nordicsemi.no>

tejlmand committed:
soc: nxp: fix typo in SoC name

Follow-up: #68778

There is a typo in nxp/kinetics/soc.yml.
The SoC `mk82f215` was added, however the correct SoC is `mk80f25615`,
which this commit corrects.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

danieldegrasse committed:
modules: mcux: introduce CONFIG_MCUX_CORE_SUFFIX

MCUX SDK depends on "core suffixes" in order to identify which core a
build is targeting on a multicore part. Previously, this information was
parsed from the CONFIG_SOC string, but with hardware model v2 this is no
longer possible. Introduce the Kconfig MCUX_CORE_SUFFIX, which multicore
SOCs can set to inform MCUX which core the build is targeting.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

manuargue-nxp committed:
tests: drivers: gpio: gpio_api_1pin: exclude hifive1

Test seems to be failing consistently on `hifive1` board.
See #69350

Signed-off-by: Manuel Argüelles <manuel.arguelles@nxp.com>

nandojve committed:
soc: atmel: Use new family prefix

The newer HWMv2 impose a different semantic in the family names. This
update from SOC_FAMILY_SAMx to SOC_FAMILY_ATMEL_SAMx to comply with.

Fixes #69046

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nordicjm committed:
soc: nordic_nrf: Rename to nordic

Renames to align with the vendor prefix

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: nordic_nrf: Rename to nordic

Aligns with the vendor prefix

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: nordic: Fix board names

Fixes some board names in documentation that wasn't updated

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: qemu: qemu_cortex_a53: Expand names

Expands names to include the SoC

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: xen: xenvm: Expand name

Expands name to include identifier

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
tests: Expand names

Expands names to include identifiers

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

jerome-pouiller committed:
soc: silabs: split in families

EXX32 does not really makes sense. Silabs tends to talk about "series
0", "series 1" and "series 2".

Note all Silabs chipsets (whatever their family) tend to share
components. So this PR introduces SOC_VENDOR_SILABS to match these
common parts.

Signed-off-by: Jérôme Pouiller <jerome.pouiller@silabs.com>

jerome-pouiller committed:
soc: silabs: align comments with soc tree

Cosmetics changes to reflect the tree organisation.

Signed-off-by: Jérôme Pouiller <jerome.pouiller@silabs.com>

jerome-pouiller committed:
soc: silabs: SOC_FAMILY_* replace SOC_GECKO_SERIES*

SOC_GECKO_SERIES* is now redundant with SOC_FAMILY_*.

Signed-off-by: Jérôme Pouiller <jerome.pouiller@silabs.com>

nordicjm committed:
soc: intel: intel_adsp: Fix issues

Fixes issues with missing protection guards and selections in
wrong files, and one case of missing bools on 2 fields

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

tejlmand committed:
soc: emsdsp: align SoC name defined in soc.yml to Kconfig SOC setting

This commit aligns the SoC name for emsdp_em4 SoC defined in Kconfig.soc
to to name in soc.yml

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

iuliana-prodan committed:
samples: update board name

Update board name and rename files for
openamp_rsc_table sample.

Signed-off-by: Iuliana Prodan <iuliana.prodan@nxp.com>

jerome-pouiller committed:
soc: silabs: introduce SOC_GECKO_SDID

Silabs hardware layer uses Silicon Die ID (SDID) for conditional
code. This patch defines SDID directly from Kconfig so Silabs HAL does
not have define it.

Signed-off-by: Jérôme Pouiller <jerome.pouiller@silabs.com>

JiafeiPan committed:
west: depend on hal_nxp PR 341

Tmp patch to be removed.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

jerome-pouiller committed:
soc: silabs: drop useless defconfigs

Currently, some soc subdirectories contains Kconfig.defconfig.<soc-name>
and Kconfig.defconfig. However:
  - Kconfig.defconfig.<soc-name> is included unconditionally by
    Kconfig.defconfig
  - each subdirectory only contains one SoC

So, it does not make sense to keep these two files.

Signed-off-by: Jérôme Pouiller <jerome.pouiller@silabs.com>

jerome-pouiller committed:
manifest: Update hal_silabs

Some changes in hal_silabs are necessary to apply this PR.

Signed-off-by: Jérôme Pouiller <jerome.pouiller@silabs.com>

JiafeiPan committed:
drivers: clock_control: ccm_rev2: align with hwm v2 update

Modified SoC configuration item name.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
drivers: pinctrl: imx: align with hwm v2

Changed SoC configuration item name.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
hwmv2: port nxp_ls1046ardb board to V2

Port NXP ls1046ardb board to hardware model v2.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

JiafeiPan committed:
hwmv2: port NXP SoC LS1046A to V2

Port NXP SoC LS1046A to hardware model v2.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

iuliana-prodan committed:
modules: mcux: remove SOC_FAMILY_NXP_ADSP

Remove SOC_FAMILY_NXP_ADSP since this is not used
anymore for the ADSP.
With HWMv2 ADSP is part of SOC_FAMILY_IMX.

Signed-off-by: Iuliana Prodan <iuliana.prodan@nxp.com>

nordicjm committed:
boards: Fix bools and selections

Fixes issues with some outstanding bools being defined and wrong
usage of soc selection

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: Documentation link fixes

Fixes missing or broken documentation links

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: Add include guards

Adds include guards to prevent contamination of bleeding
Kconfigs from irrelevent devices

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
tests/samples: Rename overlay files for renamed boards

These overlay files were not being used after board name changes
in hwmv2, rename them to have them used

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

iuliana-prodan committed:
modules: rename SOC_FAMILY_IMX

Rename SOC_FAMILY_IMX to SOC_FAMILY_NXP_IMX.

I've also kept SOC_FAMILY_IMX since there are
still legacy socs/boards that use it. When all
are ported to HWMv2 this should be removed.

Signed-off-by: Iuliana Prodan <iuliana.prodan@nxp.com>

nordicjm committed:
soc: ite: ec: it8xxx2: Fix SOC_SERIES being in wrong file

Fixes this line being in the wrong file

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
drivers: flash: it8xxx2: Add missing Kconfigs

The original hwmv1 Kconfigs for this flash driver were in the
complete wrong place and should not have been merged, which then
caused confusion when porting to hwmv2, this adds in the missing
Kconfigs to where they should have been in the first place

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

fabiobaltieri committed:
scripts: ci: check_compliance: add a check for board yml file

Add a check for board.yml file, just check for valid vendor prefixes for
now.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri committed:
gitignore: update the compliance file list

Update the list of gitignore file with the current list of compliance
checks.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri committed:
dts: vendor-prefixes: ensure all prefixes are lowercase

There's only three outliers, make sure all prefixes are lowercasel.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri committed:
dts: vendor-prefixes: sort the vendor list

Sort the list alphabetically.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri committed:
dts: vendor-prefixes: add keep-sorted markers

Add the keep sorted marker to the list so it's kept sorter by the
compliance check.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri committed:
boards: sync up the vendor tags and vendor-list

Add various board vendor prefixes to vendor-prefixes.txt and fix up all
board.yml file to only use the prefixes from that file.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

nordicjm committed:
arch: arch: kconfig: Fix wrong placement of endmenu

Fixes a bug whereby endmenu was placed before the end of the file
causes what should be ARC-only Kconfig choices to bleed into
every other architecture

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
maintainers: Fix paths

Updates paths from hwmv2 changes

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: 96boards: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: actinius: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: adafruit: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: adi: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: altera: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arduino: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: bbc: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: blue_clover: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: circuit_dojo: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: contextual_electronics: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: dragino: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: ebyte: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: google: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: holyiot: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: intel: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: khados: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: mikroe: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: olimex: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: particle: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: qemu: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: raytac: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: ronoth: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: ruuvi: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: segger: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: sparkfun: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: waveshare: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: wurth_elektronik: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: Update documentation links

Updates documentation links for shortened paths

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
samples: Fix invalid links

Fixes links that are invalid

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

JiafeiPan committed:
hwmv2: clean up arm64 soc and board empty directory

There is no legacy ARM64 SoC and Board available, so delete
the arm64 directory.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

nordicjm committed:
boards: nuvoton: numaker_pfm: Split into 2 boards

Splits up a wrongly unified board into 2 boards

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
samples/tests: Rename numaker boards

Renames boards due to hwmv2 changes

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: arm: designstart: Fix SOC_SERIES and SOC mismatch

Fixes mismatch between values and names

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: raspberry_pi: Fix SOC_SERIES and SOC mismatch

Fixes mismatch between values and names

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

fkokosinski commented on boards/starfive/visionfive2_jh7110/visionfive2_jh7110.dts:
```diff
@@ -0,0 +1,30 @@
+
+/ {
+	model = "StarFive VisionFive V2";
+	compatible = "starfive,visionfive-v2", "starfive,jh7110";
```
```suggestion
	compatible = "starfive,visionfive-v2";
```

fkokosinski commented on boards/starfive/visionfive2_jh7110/visionfive2_jh7110_defconfig:
```diff
@@ -0,0 +1,16 @@
+# Serial Drivers
+CONFIG_SERIAL=y
+CONFIG_UART_INTERRUPT_DRIVEN=y
+CONFIG_UART_NS16550=y
```
Continuing the discussion from https://github.com/zephyrproject-rtos/zephyr/pull/69359#discussion_r1502495926.

The SiFive UART is not enabled for HiFive Unmatched. You're likely referencing the `CONFIG_UART_SIFIVE_PORT_n` options - note that these are additional symbols. The Kconfig symbol for the driver itself is `CONFIG_UART_SIFIVE`, and is enabled by the `DT_HAS_SIFIVE_UART0_ENABLED` dts-generated config.

kanakshilledar replied:
Updated the file accordingly

iuliana-prodan committed:
soc: nxp: Port i.MX ADSP family to hardware model v2

Port i.MX ADSP family to HWMv2, including series:
- imx8
- imx8m
- imx8x
- imx8ulp

Signed-off-by: Iuliana Prodan <iuliana.prodan@nxp.com>

iuliana-prodan committed:
boards: nxp: Convert i.MX ADSP boards to hardware model v2

Convert i.MX ADSP boards to HMV2 including:
- imx8mp_evk
- imx8qxp_mek
- imx8qm_mek
- imx8ulp_evk

Signed-off-by: Iuliana Prodan <iuliana.prodan@nxp.com>

iuliana-prodan committed:
west: update xtensa hal

Update Xtensa HAL to get the
rename of the SoC overlay for
NXP i.MX8ULP.

Signed-off-by: Iuliana Prodan <iuliana.prodan@nxp.com>

fabiobaltieri committed:
soc: mec: rename all mec1501x reference to mec15xx

Fix a bunch of references to "mec1501x" to be named "mec15xx" instead,
which is better representative of what is supported and also matches the
soc series name and value.

Fix the clock div option name as well while at it since it apparently
applies to the whole family.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

iuliana-prodan committed:
manifest: update SOF sha for NXP HWMv2

Update SOF for NXP's new boards name and configs.

Signed-off-by: Iuliana Prodan <iuliana.prodan@nxp.com>

kanakshilledar commented:
@fkokosinski any update on this PR?

nordicjm committed:
samples/tests: Use board revisions

Updates boards so that the correct revision/revisions is/are tested

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: weact: Drop duplicate prefix

Drops the prefix since the parent folder has it already

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arm: riscv32_virtual: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: riscv_renode_virtual: Port to HWMv2

Ports the SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: altera: Rename to altr

Aligns with the vendor prefix

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: raspberry_pi: Rename to raspberrypi

Aligns with the vendor prefix

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: wurth_elektronik: Rename to we

Aligns with the vendor prefix

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: raspberry_pi: Rename to raspberrypi

Aligns with the vendor prefix

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: renode: Add documentation index

Adds a documentation index file

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: Add yaml files for boards missing revisions

Adds dedicated yaml files for board that have revisions

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

golowanow committed:
boards: intel: doc: hwmv2: Fix some renamed paths

Fix some paths affected by 'drop duplicate prefix' at folder
names in boards change.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

nordicjm committed:
tests: drivers: flash: Renamed missed board rename

Fixes a test which missed the rename of a board

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: weact: stm32g431_core: Add wrongly deleted file back

Adds a file back that was wrongly deleted in an unrelated
commit

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: Fix SOC_FAMILY name mismatches

Fixes mismatches or missing SOC_FAMILY entries

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: brcm: bcm_vk: Rename to bcnvk

Fixes a name mismatch with a family

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

fkokosinski commented:
> @fkokosinski any update on this PR?

Hey, there are some issues issues with the PR and the changes it introduces:
* you have merge commits in your PR
* your commits are not squashed properly, making bisecting problematic in the future
* the documentation fails to build with your changes
* it looks like the board is not covered by any of the areas of maintenance as defined in the `MAINTAINERS.yml` file
* it also looks like you're adding a commit with me as a co-author. Please do not do that.

Also please not that as per Zephyr Project guidelines, reviewers have 1 week to reply to the changes contributors make in their PRs etc. While it usually take less than that, sometimes (e.g. when there's a new release coming, or in general - big changes are introduced) it may take some days before reviewers can get back to your PR :)

nordicjm committed:
manifest: Update hal_silabs

Brings in a fix for a removed Kconfig name

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

golowanow committed:
hwmv2: boards: intel: adsp: Fix runner after paths renamed

Fix intel_adsp runner unable to find boards for flash after
'drop duplicate prefix' folder renames #69505

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

kanakshilledar committed:
boards: starfive: add visionfive2_jh7110 board

added VisionFive2 Board files based on JH7110 SoC.
this is basic board support. Board can be built using
`visionfive2_jh7110`

Signed-off-by: Kanak Shilledar <kanakshilledar111@protonmail.com>

pfarwsi committed:
soc: starfive: jh71xx: add jh7110 soc support

These list of files add Kconfig support for
Starfive JH7110 SOC support.

Signed-off-by: Pratik Farkase <pratik.farkase@wsisweden.com>

pfarwsi committed:
dts: riscv: starfive: add DT includes for JH7110 SOC

These list of files add basic support for StarFive
JH7110 SOC Device Tree includes for VisionFive2
board.

Signed-off-by: Pratik Farkase <pratik.farkase@wsisweden.com>

pfarwsi committed:
CODEOWNERS: update CODEOWNERS file

Update the CODEOWNERS file with StarFive
VisionFive2 board maintainers.

Signed-off-by: Pratik Farkase <pratik.farkase@wsisweden.com>

kanakshilledar committed:
MAINTAINERS: update `MAINTAINERS.yml`

updated the maintainers for Starfive Visionfive2 board.

Signed-off-by: Kanak Shilledar <kanakshilledar111@protonmail.com>

kanakshilledar commented:
Hi @fkokosinski!
Thanks for you reply

> you have merge commits in your PR

We removed all the merges and just took our commits and pushed them.  You can see our commits from `0c208eca19127a858f76381c409e30b4a963168f` commit hash. 

> your commits are not squashed properly, making bisecting problematic in the future

We feel that these re submitting the changes might fix those

> the documentation fails to build with your changes

We checked the build logs and it reported loss of connectivity. 

> it looks like the board is not covered by any of the areas of maintenance as defined in the `MAINTAINERS.yml` file

We are adding ourselves as the Maintainers for this board. Under the `RISCV` area.

> it also looks like you're adding a commit with me as a co-author. Please do not do that.

We are doing this PR as a team of 2 people. With the whole SoC and Devicetree support by @pfarwsi and board support by me.
